### PR TITLE
[stable/traefik] Adding support for Dynamic tolerations and nodeSelector

### PR DESCRIPTION
**Pull request by djsly**
Adding supports for dynamically applying a node selector to allow the pods to runs on specific nodes. See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector for more details.

Adding supports for dynamically allowing the pods to runs on tainted nodes using tolerations.
See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#taints-and-tolerations-beta-feature for more details.

### CODE REVIEW

**Comment by djsly** [comment]
/cc @lachie83 
@krancour 

Can you guys review? Thanks!

File: stable/traefik/README.md [hunk 1]
```diff
@@ -95,6 +95,8 @@ The following tables lists the configurable parameters of the Traefik chart and
 | `memoryRequest`                 | Initial share of memory requested per Traefik pod                    | `20Mi`                                    |
 | `cpuLimit`                      | CPU limit per Traefik pod                                            | `200m`                                    |
 | `memoryLimit`                   | Memory limit per Traefik pod                                         | `30Mi`                                    |
+| `nodeSelector`                  | node labels for pod assignment                            | {}                                    |
```
Comment (krancour) [inline stable/traefik/README.md]: For consistency, capitalize the first word in the description. Also use backticks around the empty map.

File: stable/traefik/README.md [hunk 2]
```diff
@@ -95,6 +95,8 @@ The following tables lists the configurable parameters of the Traefik chart and
...
+| `tolerations`                   | List of node taints to tolerate (requires Kubernetes >= 1.6) | []                                |
```
Comment (krancour) [inline stable/traefik/README.md]: For consistency, use backticks around the empty array.

File: stable/traefik/templates/deployment.yaml [hunk 3]
```diff
@@ -21,6 +21,10 @@ spec:
         heritage: "{{ .Release.Service }}"
     spec:
       terminationGracePeriodSeconds: 60
+      {{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
```
Comment (krancour) [inline stable/traefik/templates/deployment.yaml]: Very clean. I like this line. :)

**Comment by krancour** [comment]
I haven't tried this, but this looks very useful and cleanly implemented. LGTM. @lachie83 or @viglesiasce can one of you please mark this as ok to test?

Note, as always, version number should also be incremented accordingly before merge.

File: stable/traefik/README.md (hunk 1 above)
Comment (djsly) [reply stable/traefik/README.md]: Fixed

File: stable/traefik/README.md (hunk 2 above)
Comment (djsly) [reply stable/traefik/README.md]: Fixed

**Comment by djsly** [comment]
I'm not sure what is expected for the version number increment. Let me know if this is ok.

**Comment by krancour** [comment]
> I'm not sure what is expected for the version number increment. Let me know if this is ok.

I'm not sure yet either. afaik, charts are continuously (or near continuously) deployed, so 1 PR implicitly == 1 release. This means we need to be careful about incrementing the version number properly, which requires coordination with other open PRs for the same chart...

I am LGTM on this. I'll ask @viglesiasce or @lachie83 to just double-check the version bump before this gets merged.

**Comment by kachkaev** [comment]
V 1.4.0 has already been taken by https://github.com/kubernetes/charts/pull/1225 and 1.5.0 has been just mentioned by me in https://github.com/kubernetes/charts/pull/1401. So the chart version in this PR might need to change depending on the merge order.

**Comment by djsly** [comment]
@kachkaev, @krancour  Ok, I have used `version: 1.6.0` and fixed the merge conflic. Please merge the PR for upgrading to traefik 1.3.1 first!

Who can give the `/ok-to-test` ?

**Comment by djsly** [comment]
@viglesiasce please merge #1401 before this one.

**Comment by djsly** [comment]
Merge conflicts have been fixed.

**Pull request merged**